### PR TITLE
Support associative array element member access in constraint

### PR DESCRIPTION
**Pull request by kdanecki**
This is a third part of https://github.com/verilator/verilator/pull/8146. Requires https://github.com/verilator/verilator/pull/8243 to be merged before this.

It adds support for associative arrays in nested access. They are treated similarly to the arrays. That is, we write_var for each element that is constrained, e.g. `m_holder.m_string_assoc[m_key].m_x` -> `"m_holder.m_string_assoc.<value of m_key>.m_x"`and `m_assoc_nested[123][1].m_x` -> `"m_assoc_nested.123.1.m_x"`.

### CODE REVIEW

File: src/V3Randomize.cpp [hunk 1]
```diff
@@ -2252,6 +2262,15 @@ class ConstraintExprVisitor final : public VNVisitor {
                     fmt = (m_structSel ? "%" : "#x%")
                           + std::to_string(VL_WORDS_I(actual_width) * 8) + "x";
                 }
+                if (m_nestedAccess) {
+                    AstNodeExpr* const bitp = nodep->bitp()->cloneTreePure(false);
+                    AstNodeExpr* const bitFormatp
+                        = new AstSFormatF{bitp->fileline(), fmt, false, bitp};
+                    AstSFormatF* const herep
+                        = new AstSFormatF{nodep->fileline(), "%s.%s", false, bitFormatp};
```
Comment (wsnyder) [inline src/V3Randomize.cpp:2270]: Use addStringNamePart, or modify it so you can.

File: src/V3Randomize.cpp
Comment (wsnyder) [inline src/V3Randomize.cpp:1]: Claude (maybe wrong):

 visit(AstAssocSel*)'s new addStringNamePart() helper is called from two branches — a  VarRef string-key branch (variable key, gets a real name) and a CvtPackString branch  specifically for literal string keys (e.g. m_assoc["abc"]). In the literal-key case,  nodep->bitp() is the AstCvtPackString conversion node itself, which doesn't override name(),  so addVarNamePart() receives an empty name — exactly the trigger condition for #8243's buggy  "multiple expressions" tracking. Since a literal string key is the single most idiomatic way  to index a string-keyed associative array in a constraint (m_cfg["mode"].value == 1), this PR  turns a previously narrow edge case (a non-trivial arithmetic array index) into something that  fires on ordinary, expected usage.

  This PR's own test only reaches the VarRef branch (m_string_assoc[m_key], a variable key)  inside the actual constraint — it reads m_string_assoc["abc"] back at the plain SystemVerilog  level, not inside a constraint expression — so the new CvtPackString/literal-key path this PR  adds is never exercised by its own test suite.

**Review by wsnyder — CHANGES_REQUESTED** [review]
